docs/DeveloperGuide.md

```diff
@@ -2,33 +2,166 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+**`HdBuy`** has [`HdBuy`](https://https://github.com/AY2021S2-CS2113-F10-1/tp/blob/master/src/main/java/seedu/hdbuy/HdBuy.java). It is responsible for,
+* At app launch: Loads previously shortlisted resale flats, if any, and starts receiving user commands.
+* At shut down: Clears search history and saves shortlisted resale flats to local file.
 
+The rest of the App consists of:
+
+* [**`Api`**](#api-component): Retrieves data on resale flats from server.
+* [**`Command`**](#command-component): The command executor.
+* [**`Parser`**](#parser-component): Translate user input to valid commands to be executed.
+* [**`Ui`**](#ui-component): Communicates with user via messages.
+* [**`Data`**](#data-component): Contains shortlist, user input history and temporarily tracks search history.
+* [**`Storage`**](#storage-component): Reads shortlisted units from, and writes shortlisted units to, the text file.
+* [**`Common`**](#common-component): Models of objects used internally.
+
+### Api component
+
+![Structure of the Api Component](diagrams/ApiClass.png)
```

Comment:
Is this format correct? I believe the label arrow can be omitted.
![image](https://user-images.githubusercontent.com/61216523/113233687-60d08d80-92d2-11eb-81c0-b84f534fcdf7.png)